### PR TITLE
App publication  requests.

### PR DESCRIPTION
**Pull request by slr71**

### CODE REVIEW

File: src/apps/service/apps/de/validation.clj [hunk 1]
```diff
@@ -51,11 +51,9 @@
   "A tool is publishable if the authenticated user has ownership of the tool and the Docker container that the tool
    uses is in one of the trusted Docker registries."
```
Comment (psarando) [inline src/apps/service/apps/de/validation.clj]: Looks like this function's docs need to be updated too?

File: src/apps/service/apps/de/validation.clj [hunk 2]
```diff
@@ -88,6 +86,16 @@
...
 
+(defn uses-tools-in-untrusted-registries?
+  "Determines whether or not any of the tools used by an app are in an untrusted Docker repository."
+  [app-id]
+  (let [trusted-registries     (set (config/trusted-registries))
+        public-tool-ids        (set (perms-client/get-public-tool-ids))
+        private-tool?          #(not (contains? public-tool-ids (:id %)))
+        get-registry           (comp first #(string/split % #"/" 2) :image_name)
+        in-untrusted-registry? #(not (contains? trusted-registries (get-registry %)))]
+    (some (every-pred in-untrusted-registry? private-tool?) (get-app-tools app-id))))
```
Comment (psarando) [inline src/apps/service/apps/de/validation.clj]: 👍

File: src/apps/service/apps/de/validation.clj (hunk 1 above)
Comment (slr71) [reply src/apps/service/apps/de/validation.clj]: Good catch. Thanks!

**Comment by slr71** [comment]
Thanks for the review!

**Pull request merged**